File: 00Arrays/03tournamentwinner.py
def tournamentWinner( competitions: list, results: list ) -> str:
    currentBestTeam = ""
    scores = { currentBestTeam : 0}

    for competition, result in zip(competitions, results):
        homeTeam = competition[0]
        awayTeam = competition[1]

        winnerTeam = homeTeam if result == 1 else awayTeam

        updateScores(scores, winnerTeam)
        print(scores)

        if scores[winnerTeam] > scores[currentBestTeam]:
            currentBestTeam = winnerTeam
    
    return currentBestTeam

def updateScores( scores: dict, winner_team: str ) -> None:
    if winner_team not in scores:
        scores[winner_team] = 0
    scores[winner_team] += 3
    return

if __name__ == "__main__":
    competitions = [
        ["HTML", "C#"],
        ["C#", "Python"],
        ["Python", "HTML"]
    ]
    results = [0,0,1]

    print(tournamentWinner(competitions, results))

File: 00Arrays/03tournamentwinner.js
main();

function main() {
    let competitions, results;
    competitions = [
        ["HTML", "C#"],
        ["C#", "Python"],
        ["Python", "HTML"]
    ];

    results = [0,0,1];
    console.log(tournamentWinner(competitions, results));
}

function tournamentWinner( competitions, results ) {
    let currentBestTeam = "";
    let scores = {};
    scores[currentBestTeam] = 0;

    for ( let i = 0; i < competitions.length; i++ ){

        let homeTeam = competitions[i][0],
            awayTeam = competitions[i][1],
            result = results[i];
        
        let winnerTeam = result == 1 ? homeTeam : awayTeam;

        updateScores(scores, winnerTeam);

        if ( scores[winnerTeam] > scores[currentBestTeam] )
            currentBestTeam = winnerTeam;

    }

    return currentBestTeam;
}

function updateScores( scores, winnerTeam ){
    if ( scores[winnerTeam] == undefined ) scores[winnerTeam] = 0;
    scores[winnerTeam] += 3;
    return;
}